### ISSUE TITLE
[2.3.6 & 2.4.1] Submitting invalid create account form leaves the submit button disabled

### ISSUE THREAD

**Issue by hostep** [magento/magento2#30513]
### Preconditions (*)
1. Magento 2.3.6 and 2.4.1
1. Confirmed on 2.4.1 composer installation
 Note(16 oct 2020): on git `2.4-develop` branch it possibly not reproducible because the branch is not synced with release code yet

#### Steps to reproduce (*)
1. Install clean Magento 2.3.6 or 2.4.1
1. Go to the create a new account form
1. Press the 'Create an Account' button without filling in any fields
1. Notice that the button turns inactive/disabled

#### Expected result (*)
1. Button should stay active/enabled

#### Actual result (*)
1. Button stays disabled, even after filling in all required fields, so you can't submit the form

#### Steps to reproduce (*)
1. Go to Create New Customer Account
1. Fill form but do some mistake e.g. in password fields. Example:
 -- password/Confirm password is not equal
 -- password/Confirm password is not long enough
1. Click "Create an Account" button
1. Result: Form validation is failed and "Create an Account" button is not clickable(disabled)
1. Correct/fix a mistake(s)

### Expected result (*)
1. Button should stay active/enabled
1. User should be able to create new account

### Actual result (*)
1. Button stays disabled, even after filling in all required fields with valid/correct data
1. Customer is NOT able to create new Account

![30513_p1](https://user-images.githubusercontent.com/3409584/96278482-2a5cbb80-0f9b-11eb-89ee-fd4bef2c20ad.gif)
### Discussion

This is pretty bad because when people make a single mistake on a form, they can no longer submit the form after correcting their mistake, they need to refresh the form and loose all the data they entered and enter them again.

Maybe other forms are affected as well, not tested yet...
##  

Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.
 - < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
 - [x] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
 - [ ] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
 - [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
 - [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

| Patch is released in MQP|borderStyle=dashed|borderColor=#cccccc|titleBGColor=#dddddd|bgColor=#e3ffea |
| --- |
| 
* **MQP release:**
 1.0.11
* **Patch ID**
MC-38509
* **Are there any additional actions required after the patch installation to make it work?**
 N/A
* **Compatible with Magento versions:**
Magento Commerce and Magento Commerce Cloud 2.3.6, 2.4.1
[Check using Patch Checker tool](http://web2.sparta.corp.magento.com/dev/support/tools/patchchk/?patch_id=MC-38509)

* **Fix version**
 2.4.2 |

**Comment by hostep** [comment]
Update: just tested Magento 2.4.1 and it also has this problem, so updated title and details in opening post.

**Comment by sdzhepa** [comment]
Hello @hostep 

I am also checking this issue on 2.4.1.
And noticed that the button became unblocked as soon as a user filled email field

**Comment by hostep** [comment]
Hi @sdzhepa : just tested again on 2.4.1 and I can't confirm, the button stays disabled.

I did some quick git bisecting between 2.4.0 and 2.4.1 and it says this commit causes the bug: https://github.com/magento/magento2/commit/02b87f6aa277cf34f374cb4aa6782569185296da

**Comment by hostep** [comment]
Hmm, something fishy is going on, it works in 2.4.1 when you fill in the email field as last one (but not on 2.3.6) but it fails when you fill in the password field or your last name as last one.

There is definitely something weird going on.

**Comment by hostep** [comment]
It does work correctly on the 2.4-develop branch, but I don't know if the 2.4.1 code is already merged in there ....

Anyway, have to leave, time for the weekend to start! 🙂

**Comment by sdzhepa** [comment]
@hostep  
After little-bit deep investigation, i found steps to reproduce it.
Thank you for report!

If you not mind i will update the description and steps to reproduce

Steps:
1. Go to Create New Customer Account
2. Fill form but do some mistake e.g. in password fields. Example: 
-- password/Conform password is not equal 
-- password/Conform password is not long enough 
3. Click "Create an Account" button
4. Result: Form validation is failed and  "Create an Account" button is not clickable(disabled) 
5. Correct/fix a mistake(s)

### Expected result (*)
1. Button should stay active/enabled
2. User should be able to create new account

### Actual result (*)
1. Button stays disabled, even after filling in all required fields with valid/correct data
2. Customer is NOT able to create new Account

![30513_p1](https://user-images.githubusercontent.com/3409584/96278482-2a5cbb80-0f9b-11eb-89ee-fd4bef2c20ad.gif)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-38509` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by mrtuvn** [comment]
oh we have regression bug for 2.4.1

**Comment by mrtuvn** [comment]
even use latest code in 2.4-develop i still can reproduce this issue ! not sure what pieces of code merged after Note(16 oct 2020) as your report or this issue already exists before that and nobody notice

**Comment by mrtuvn** [comment]
https://github.com/magento/magento2/commit/e212f94672953ab4ad97ceb178f9b1e956fbf951
seem related

**Comment by chiranjeevi-cj** [comment]
@magento I am working on this

**Comment by chiranjeevi-cj** [comment]
@magento add to contributors team

**Comment by chiranjeevi-cj** [comment]
@magento I am working on this.

**Comment by sdzhepa** [comment]
@hostep @chiranjeevi-cj @mrtuvn 
jFYI

The internal team is working on the fix for this issue.
Internal PR is ready and in the code review stage

**Comment by hostep** [comment]
@sdzhepa: very good, thanks!

I just discovered another case of this issue, but you guys might have found it already as well:
1. Go to the Create new account form
2. Fill in an invalid email address in the subscribe to newsletter block in the footer
3. Click the Subscribe button
4. Notice the button is disabled
5. Correct the email address it is valid
6. Notice that the subscribe button is still disabled

This only happens on the create a new account form, the newsletter subscribe button acts correctly on other pages.

**Comment by sdzhepa** [comment]
The issue was closed by mistake.
at this moment Magento team has fixed it, internal PR is ready(CR and QA passed) but not merged into the main code-base.

Sorry for inconvenience

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-38509` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hostep** [comment]
It looks like Magento released a patch for this issue over here: https://magento.com/tech-resources/download (search for `MC-38509`)

**Comment by hostep** [comment]
I can confirm this patch fixes the problem on Magento 2.3.6 (we aren't using Magento 2.4.x yet, but I assume it will work as well).

For the people who install patches using some composer patch plugin, here's our adapted version:

[MC-38509-Customer.txt](https://github.com/magento/magento2/files/5443734/MC-38509-Customer.txt) (rename from `.txt` to `.diff`)

And its definition:
```json
        "magento/module-customer": {
            "MC-38509 - Create New Customer Account patch": "vendor_patches/MC-38509-Customer.diff"
        }
```

Thanks to Magento/Adobe devs for picking up this so quickly and thanks to my colleague @sorinbarnutiu for discovering the bug in the first place!

**Comment by hostep** [comment]
This probably got closed because the fix is in the `2.4-develop` branch by now. Here's the commit: https://github.com/magento/magento2/commit/76b9492952301f7f4a7b888bac9cdc16d2568d9c

**Comment by sdzhepa** [comment]
The related internal Jira ticket `MC-38509` was closed as `Fixed`.
Related commits can be found by search https://github.com/magento/magento2/search?q=MC-38509&type=commits

**Comment by lucianopeluso10** [comment]
The patch is not working in 2.4.1
